src/config.rs

```diff
@@ -62,12 +69,20 @@ pub fn get_configs(config_path: PathBuf) -> Result<Vec<CornerConfig>> {
         .with_context(|| format!("could not open the file {}", path.display()))?;
     let mut config_content = String::new();
     config_file.read_to_string(&mut config_content)?;
-    toml::from_str::<Config>(config_content.as_str())
-        .map_err(|error| -> Error { error.into() })
-        .with_context(|| format!("could not parse {}", path.display()))
-        .map(|item| {
-            item.into_iter()
-                .map(|(_key, value)| value)
-                .collect::<Vec<_>>()
-        })
+    toml::from_str::<Config>(config_content.as_str()).map(|item| {
+        item.into_iter()
+            .map(|(key, value)| {
+                if key.ends_with(".output") || value.enter_command.len() != 0 || value.exit_command.len() != 0 {
+                    Ok(value)
+                } else {
+                    Err(Error::msg(format!(
+                        "You must provide either an `exit_command` or an `enter_command` for key `{}`",
+                        key
+                    )))
```

Comment:
Is there a reason why we need to check whether the key ends with `output`? The value is a `CornerConfig` so any key with `.output` is `CornerConfig.output` or am I mistaken?

```suggestion
                if value.enter_command.is_empty() && value.exit_command.is_empty() {
                    bail!(
                        "You must provide either an `exit_command` or an `enter_command` for key `{}`",
                        key
                    )
                }
                Ok(value)
```

🤏 Also little nitpick on making it easier to read and also I prefer seeing the expected value at the bottom and short circuiting errors above. 😅

Reply:
> Is there a reason why we need to check whether the key ends with output? The value is a CornerConfig so any key with .output is CornerConfig.output or am I mistaken?

You're absolutely right, my bad, I had extraneous speech marks around "left.output" in my config
![image](https://user-images.githubusercontent.com/34243578/198888827-370972ec-aaaf-4a4b-92e3-1fb0c3be9c2f.png)